apps/meteor/client/views/room/modals/E2EEModals/ResetKeysE2EEModal.tsx

```diff
@@ -0,0 +1,62 @@
+import { Box, Modal } from '@rocket.chat/fuselage';
+import { ExternalLink } from '@rocket.chat/ui-client';
+import type { ReactElement } from 'react';
+import React from 'react';
+import { Trans, useTranslation } from 'react-i18next';
+
+import GenericModal from '../../../../components/GenericModal';
+import { dispatchToastMessage } from '../../../../lib/toast';
+import { useE2EEResetRoomKeyRoom } from '../../hooks/useE2EEResetRoomKey';
+
+const E2EE_RESET_KEY_LINK = 'https://go.rocket.chat/';
+
+type ResetKeysE2EEModalProps = {
+	roomType: string;
+	roomId: string;
+	onCancel: () => void;
+};
+
+const ResetKeysE2EEModal = ({ roomType, roomId, onCancel }: ResetKeysE2EEModalProps): ReactElement => {
+	const { t } = useTranslation();
+	const resetRoomKeyMutation = useE2EEResetRoomKeyRoom();
+
+	const handleResetRoomKey = () => {
+		resetRoomKeyMutation.mutate(
+			{ roomId },
+			{
+				onSuccess: () => {
+					dispatchToastMessage({ type: 'success', message: t('E2E_reset_encryption_keys_success') });
+				},
+				onError: () => {
+					dispatchToastMessage({ type: 'error', message: t('E2E_reset_encryption_keys_error') });
+				},
+				onSettled: () => {
+					onCancel();
+				},
+			},
+		);
+	};
+
+	return (
+		<GenericModal
+			icon={<Modal.Icon color='danger' name='key' />}
+			title={t('E2E_reset_encryption_keys')}
+			variant='danger'
+			confirmText={t('E2E_reset_encryption_keys')}
+			dontAskAgain={<Modal.FooterAnnotation>{t('This_action_cannot_be_undone')}</Modal.FooterAnnotation>}
+			onCancel={onCancel}
+			onConfirm={handleResetRoomKey}
+			onDismiss={() => undefined}
+		>
+			<Box mbe={16} is='p'>
+				<Trans i18nKey='E2E_reset_encryption_keys_modal_description' tOptions={{ roomType }}>
+					Resetting E2EE keys is only recommend if no {roomType} member has a valid key to regain access to the previously encrypted
```

Comment:
why do you need to put the whole content again if it should come from the translation? 🤔

Reply:
Yep, due to the translation positioning arguments

Reply:
https://react.i18next.com/legacy-v9/step-by-step-guide#using-the-trans-component

Reply:
Oh so yo may need to do this on the button as well to fix the translation? 👀 or is that a diff issue?

Reply:
Diff issue, I should have use the `t`function for the button, I'm fixing it